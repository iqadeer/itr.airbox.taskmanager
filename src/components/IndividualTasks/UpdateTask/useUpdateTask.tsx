import { ChangeEventHandler, useState, FormEventHandler } from 'react';
import { AbxTask } from '../../../shared/models/AbxTask';
import { APIService } from '../../../api/ApiService';

export type AbxTaskUpdateResponse = {
  data: AbxTask;
};

const useUpdateTask = () => {
  const isUpdateTaskStateValid = () => {
    const error: AbxTask = {
      abxTaskId: updateTaskState.abxTaskId ? '' : 'abxTaskId is required',
      organisationTaskId: '',
      organisationId: '',
      priority: '',
      taskStatus: '',
      assignedTo: '',
      timeStamp: '',
      latitude: '',
      longitude: '',
      taskSummary: '',
      taskDescription: '',
    };

    setUpdateTaskStateError(error);
    return updateTaskState.abxTaskId;
  };

  const handleSubmit: FormEventHandler<HTMLFormElement> = async (e) => {
    e.preventDefault();
    if (isUpdateTaskStateValid()) {
      try {
        const result = await APIService.put<AbxTask, AbxTaskUpdateResponse>(
          `tasks/${updateTaskState.abxTaskId}`,
          updateTaskState,
          'token'
        );
        setSnackBar({ message: JSON.stringify(result.data), severity: 'success' });
        setOpenSnackBar(true);
        setUpdateTaskState({
          abxTaskId: '',
          organisationTaskId: '',
          organisationId: '',
          priority: '',
          taskStatus: '',
          assignedTo: '',
          timeStamp: '',
          latitude: '',
          longitude: '',
          taskSummary: '',
          taskDescription: '',
        });
        console.log(result.data);
      } catch (e) {
        setSnackBar({ message: `Error updating task: ${e}`, severity: 'error' });
        setOpenSnackBar(true);
      }
    }
  };

  const [openSnackBar, setOpenSnackBar] = useState<boolean>(false);
  const [snackBar, setSnackBar] = useState<{
    message: string;
    severity: 'error' | 'success';
  }>({
    message: '',
    severity: 'success',
  });
  const closeSnackBar = (): void => {
    setOpenSnackBar(false);
  };

  const [updateTaskState, setUpdateTaskState] = useState<AbxTask>({
    abxTaskId: '',
    organisationTaskId: '',
    organisationId: '',
    priority: '',
    taskStatus: '',
    assignedTo: '',
    timeStamp: '',
    latitude: '',
    longitude: '',
    taskSummary: '',
    taskDescription: '',
  });

  const [updateTaskStateError, setUpdateTaskStateError] = useState<AbxTask>({
    abxTaskId: '',
    organisationTaskId: '',
    organisationId: '',
    priority: '',
    taskStatus: '',
    assignedTo: '',
    timeStamp: '',
    latitude: '',
    longitude: '',
    taskSummary: '',
    taskDescription: '',
  });

  const handleChange: ChangeEventHandler<HTMLInputElement> = ({ target: { name, value } }) => {
    setUpdateTaskState({ ...updateTaskState, [name]: value });
    const errorMessage = !value ? `${name} is required` : '';
    setUpdateTaskStateError({ ...updateTaskStateError, [name]: errorMessage });
  };

  return {
    handleSubmit,
    handleChange,
    openSnackBar,
    closeSnackBar,
    message: snackBar.message,
    severity: snackBar.severity,
    updateTaskState,
    updateTaskStateError,
  };
};

export default useUpdateTask;
