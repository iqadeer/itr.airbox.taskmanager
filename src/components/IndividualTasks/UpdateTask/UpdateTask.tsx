import React from 'react';
import { Grid, Button, TextField, Snackbar } from '@material-ui/core';
import useStyles from './useStyles';
import UpdateIcon from '@material-ui/icons/Update';
import MuiAlert, { AlertProps } from '@material-ui/lab/Alert';
import useUpdateTask from './useUpdateTask';

const Alert = (props: AlertProps) => {
  return <MuiAlert elevation={16} variant='filled' {...props} />;
};
const UpdateTask = () => {
  const classes = useStyles({});
  const {
    handleSubmit,
    handleChange,
    openSnackBar,
    closeSnackBar,
    message,
    severity,
    updateTaskState: state,
    updateTaskStateError: error,
  } = useUpdateTask();
  return (
    <>
      <form className={classes.form} onSubmit={handleSubmit} noValidate>
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              autoFocus
              error={!!error.abxTaskId}
              placeholder='Abx task id'
              autoComplete='Abx task id'
              name='abxTaskId'
              variant='outlined'
              required
              fullWidth
              id='abxTaskId'
              value={state.abxTaskId}
              onChange={handleChange}
              label='Abx task id'
              helperText={error.abxTaskId}
            />
          </Grid>
        </Grid>
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='Organisation task id'
              autoComplete='Organisation task id'
              name='organisationTaskId'
              variant='outlined'
              fullWidth
              id='organisationTaskId'
              value={state.organisationTaskId}
              onChange={handleChange}
              label='Organisation task id'
            />
          </Grid>
        </Grid>
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='Organisation id'
              autoComplete='Organisation id'
              name='organisationId'
              variant='outlined'
              fullWidth
              id='organisationId'
              value={state.organisationId}
              onChange={handleChange}
              label='Organisation id'
            />
          </Grid>
        </Grid>
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='Priority'
              autoComplete='Priority'
              name='priority'
              variant='outlined'
              fullWidth
              id='priority'
              value={state.priority}
              onChange={handleChange}
              label='Priority'
            />
          </Grid>
        </Grid>
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='Task status'
              autoComplete='Task status'
              name='taskStatus'
              variant='outlined'
              fullWidth
              id='taskStatus'
              value={state.taskStatus}
              onChange={handleChange}
              label='Task status'
            />
          </Grid>
        </Grid>
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='Assigned to'
              autoComplete='Assigned to'
              name='assignedTo'
              variant='outlined'
              fullWidth
              id='assignedTo'
              value={state.assignedTo}
              onChange={handleChange}
              label='Assigned to'
            />
          </Grid>
        </Grid>
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='latitude'
              autoComplete='latitude'
              name='latitude'
              variant='outlined'
              fullWidth
              id='latitude'
              value={state.latitude}
              onChange={handleChange}
              label='latitude'
            />
          </Grid>
        </Grid>{' '}
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='longitude'
              autoComplete='longitude'
              name='longitude'
              variant='outlined'
              fullWidth
              id='longitude'
              value={state.longitude}
              onChange={handleChange}
              label='longitude'
            />
          </Grid>
        </Grid>{' '}
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='Task summary'
              autoComplete='Task summary'
              name='taskSummary'
              variant='outlined'
              fullWidth
              id='taskSummary'
              value={state.taskSummary}
              onChange={handleChange}
              label='Task summary'
            />
          </Grid>
        </Grid>{' '}
        <Grid container justify='center' alignItems='center' spacing={2}>
          <Grid item xs={12} md={6}>
            <TextField
              placeholder='Task description'
              autoComplete='Task description'
              name='taskDescription'
              variant='outlined'
              fullWidth
              id='taskDescription'
              value={state.taskDescription}
              onChange={handleChange}
              label='Task description'
            />
          </Grid>
        </Grid>
        <Grid container justify='center' alignItems='center'>
          <Grid item xs={12} md={6}>
            <Button
              type='submit'
              variant='contained'
              color='primary'
              startIcon={<UpdateIcon />}
              className={classes.submit}>
              Update task
            </Button>
          </Grid>
        </Grid>
      </form>
      <Snackbar
        autoHideDuration={3000}
        anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
        key={`top,center`}
        open={openSnackBar}
        onClose={closeSnackBar}>
        <Alert onClose={closeSnackBar} severity={severity}>
          {message}
        </Alert>
      </Snackbar>
    </>
  );
};

export default UpdateTask;
