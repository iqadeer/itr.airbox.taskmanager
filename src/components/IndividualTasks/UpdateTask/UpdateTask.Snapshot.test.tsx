import React from 'react';
import renderer from 'react-test-renderer';
import UpdateTask from './UpdateTask';

it('should have correct button label', () => {
  const task = renderer.create(<UpdateTask></UpdateTask>);
  expect(task).toMatchSnapshot();
});
