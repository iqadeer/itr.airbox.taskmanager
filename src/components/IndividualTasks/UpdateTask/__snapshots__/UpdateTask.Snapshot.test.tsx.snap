// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have correct button label 1`] = `
<form
  className="makeStyles-form-5"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="abxTaskId"
          id="abxTaskId-label"
        >
          Abx task id
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="Abx task id"
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="abxTaskId"
            name="abxTaskId"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Abx task id"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                Abx task id
                 *
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="organisationTaskId"
          id="organisationTaskId-label"
        >
          Organisation task id
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="Organisation task id"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="organisationTaskId"
            name="organisationTaskId"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Organisation task id"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                Organisation task id
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="organisationId"
          id="organisationId-label"
        >
          Organisation id
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="Organisation id"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="organisationId"
            name="organisationId"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Organisation id"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                Organisation id
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="priority"
          id="priority-label"
        >
          Priority
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="Priority"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="priority"
            name="priority"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Priority"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                Priority
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="taskStatus"
          id="taskStatus-label"
        >
          Task status
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="Task status"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="taskStatus"
            name="taskStatus"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Task status"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                Task status
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="assignedTo"
          id="assignedTo-label"
        >
          Assigned to
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="Assigned to"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="assignedTo"
            name="assignedTo"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Assigned to"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                Assigned to
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="latitude"
          id="latitude-label"
        >
          latitude
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="latitude"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="latitude"
            name="latitude"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="latitude"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                latitude
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
   
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="longitude"
          id="longitude-label"
        >
          longitude
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="longitude"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="longitude"
            name="longitude"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="longitude"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                longitude
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
   
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="taskSummary"
          id="taskSummary-label"
        >
          Task summary
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="Task summary"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="taskSummary"
            name="taskSummary"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Task summary"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                Task summary
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
   
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="taskDescription"
          id="taskDescription-label"
        >
          Task description
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="Task description"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="taskDescription"
            name="taskDescription"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Task description"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-12"
            >
              <span>
                Task description
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-8 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 10.12h-6.78l2.74-2.82c-2.73-2.7-7.15-2.8-9.88-.1-2.73 2.71-2.73 7.08 0 9.79s7.15 2.71 9.88 0C18.32 15.65 19 14.08 19 12.1h2c0 1.98-.88 4.55-2.64 6.29-3.51 3.48-9.21 3.48-12.72 0-3.5-3.47-3.53-9.11-.02-12.58s9.14-3.47 12.65 0L21 3v7.12zM12.5 8v4.25l3.5 2.08-.72 1.21L11 13V8h1.5z"
              />
            </svg>
          </span>
          Update task
        </span>
      </button>
    </div>
  </div>
</form>
`;
